### PR TITLE
[speex] Use native autotools for all triplets

### PR DESCRIPTION
**Pull request by dg0yt**
Alternative to https://github.com/microsoft/vcpkg/pull/26487#issuecomment-1225182426.

- #### What does your PR fix?
  Use the native build system for all triplets. Fixes mingw builds.
  Adds speexdsp dependency/feature control.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

File: ports/speex/speexdsp-control.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
+diff --git a/configure.ac b/configure.ac
+index b813148..3ad9531 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -254,7 +254,11 @@ AC_SUBST(FFT_PKGCONFIG)
+ 
+ AM_CONDITIONAL(BUILD_VORBIS_PSY, [test "x$vorbis_psy" = "xyes"])
+ 
+-PKG_CHECK_MODULES([SPEEXDSP], [speexdsp], [AC_DEFINE([USE_SPEEXDSP], [], [Use SpeexDSP library])], [speexdsp_failed=yes])
```
Comment (JackBoosY) [inline ports/speex/speexdsp-control.patch:9]: Why set this as a feature?

Comment (dg0yt) [reply ports/speex/speexdsp-control.patch:9]: I will remove it. speexdsp is only used by the executables which aren't built by vcpkg.

**Review by JackBoosY — APPROVED** [review]

File: ports/speex/fix-vla-check.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/configure.ac b/configure.ac
+index d94c2fe..b813148 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -37,7 +37,7 @@ AC_C_RESTRICT
+ 
+ 
+ AC_MSG_CHECKING(for C99 variable-size arrays)
+-+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[
++AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[
```
Comment (dan-shaw) [inline ports/speex/fix-vla-check.patch:10]: What does this patch do?

Comment (dg0yt) [reply ports/speex/fix-vla-check.patch:10]: It fixes an upstream typo which broke that test. Cf. https://github.com/xiph/speex/pull/20.

**Pull request merged**